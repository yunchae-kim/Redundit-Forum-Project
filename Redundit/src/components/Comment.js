import { Link, useHistory } from "react-router-dom";
import { useEffect, useState } from "react";
import { Divider } from "@blueprintjs/core";
import userIcon from "../resources/userIcon.png";
import "../style/components/CommentList.css";
import { getHashtag } from "../utility/axiosHashtags";

const Comment = ({ comment, userId, deleteCmt }) => {
  const history = useHistory();
  const [tags, setTags] = useState([]);

  const editComment = () => {
    history.push({ pathname: `/editcomment/${comment.idcomment}`, state: { comment, hashtags: tags } });
  };

  useEffect(() => {
    getHashtag(comment.idcomment, "comment").then((tagsFetched) => {
      setTags(tagsFetched);
    });
  }, []);

  const hashTagsList = tags.map((h) => (
    <Link to={{ pathname: `/hashtag/${h.hashtag}`, state: { hashtag: h.hashtag } }} key={h.hashtag}>
      #
      {h.hashtag}
      {" "}
    </Link>
  ));

  return (
    <div className="Post-comment">
      <div className="Post-comment-name">
        <img src={userIcon} alt="" className="Post-comment-icon" />
        {comment.status === 1
          ? <Link to={{ pathname: `/user/${comment.iduser}`, state: { userId: comment.iduser } }}><span className="CW-chatentry-name">{comment.username}</span></Link>
          : (
            <span className="CW-chatentry-name">
              {comment.username}
              {" "}
              (deactivated)
            </span>
          )}
        <span className="Post-comment-time">{comment.date}</span>
      </div>
      <div className="Post-comment-content">{hashTagsList}</div>
      <div className="Post-comment-content">{comment.content}</div>
      { comment.iduser === userId && (
      <div className="Post-comment-modify">
        <button type="submit" onClick={() => editComment()}>Edit Comment</button>
        <Divider />
        <button type="submit" onClick={() => deleteCmt(comment.idcomment)}>Delete Comment</button>
      </div>
      ) }
    </div>
  );
};

export default Comment;
