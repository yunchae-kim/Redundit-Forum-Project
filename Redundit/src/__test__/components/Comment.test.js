import { render, screen, waitFor } from "@testing-library/react";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom";
import { BrowserRouter } from "react-router-dom";

import Comment from "../../components/Comment";

const testComment = {
  iduser: 1,
  idpost: 1,
  content: "this is a new comment",
  status: false,
  flagged: false,
  image: "https://source.unsplash.com/random",
  audio: "https://source.unsplash.com/random",
  video: "https://source.unsplash.com/random",
};

test("test should render Comment component", () => {
  render(
    <BrowserRouter>
      <Comment comment={testComment} iduser={1} idpost={1} />
    </BrowserRouter>,
  );
  const title = screen.getByText("this is a new comment");
  expect(title).toBeInTheDocument();
});

test("test should CommentList match snapshot", () => {
  const tree = renderer.create(
    <BrowserRouter>
      <Comment comment={testComment} iduser={1} idpost={1} />
    </BrowserRouter>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
