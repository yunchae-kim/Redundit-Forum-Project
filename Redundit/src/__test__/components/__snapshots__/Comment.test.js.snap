// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test should CommentList match snapshot 1`] = `
<div
  className="Post-comment"
>
  <div
    className="Post-comment-name"
  >
    <img
      alt=""
      className="Post-comment-icon"
      src="userIcon.png"
    />
    <span
      className="CW-chatentry-name"
    >
       
      (deactivated)
    </span>
    <span
      className="Post-comment-time"
    />
  </div>
  <div
    className="Post-comment-content"
  />
  <div
    className="Post-comment-content"
  >
    this is a new comment
  </div>
</div>
`;
